language/move-core/types/src/language_storage.rs

```diff
@@ -151,3 +151,87 @@ impl Display for TypeTag {
         }
     }
 }
+
+// =================================================================================================
+// Type definition with free type variables
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub enum Type {
```

Comment:
Probably will need better name for this and `StructType`. Seems to bring so much confusion.

Reply:
I would strongly prefer that we use `normalized::Type` for this instead of adding yet another definition if we can... Though it might make sense to move `normalized::Type` out of the VM crate and into `types`.